src/components/numeral-date/numeral-date.component.tsx

```diff
@@ -244,7 +378,7 @@ const NumeralDate = ({
                       disabled={disabled}
                       readOnly={readOnly}
                       placeholder={datePart}
-                      value={dateValue[datePart]}
+                      value={dateValue[datePart as keyof NumeralDateObject]}
```

Comment:
Same as the type assertion comment on `Object.fromEntries`

Reply:
response in the comment above